connectors/src/connectors/github/index.ts

```diff
@@ -459,6 +462,218 @@ export async function retrieveGithubReposTitles(
+      },
+    }),
+    GithubCodeDirectory.findAll({
+      where: {
+        connectorId: c.id,
+        internalId: codeDirectoryIds,
+      },
+    }),
+    GithubCodeFile.findAll({
+      where: {
+        connectorId: c.id,
+        documentId: codeFileIds,
+      },
+    }),
+  ]);
+
+  // Constructing Nodes for Full Repo
+  fullRepoIds.forEach((repoId) => {
+    const repo = uniqueRepos[repoId];
+    if (!repo) {
+      return;
+    }
+    nodes.push({
+      provider: c.type,
+      internalId: repoId.toString(),
+      parentInternalId: null,
+      type: "folder",
+      title: repo.name,
+      titleWithParentsContext: `[${repo.name}] - Full repository`,
+      sourceUrl: repo.url,
+      expandable: true,
+      permission: "read",
+      dustDocumentId: null,
+      lastUpdatedAt: null,
+    });
+  });
+
+  // Constructing Nodes for All Issues and All Discussions
+  allIssuesFromRepoIds.forEach((repoId) => {
+    const repo = uniqueRepos[repoId];
+    if (!repo) {
+      return;
+    }
+    nodes.push({
+      provider: c.type,
+      internalId: `${repoId}-issues`,
+      parentInternalId: repoId.toString(),
+      type: "database",
+      title: "Issues",
+      titleWithParentsContext: `${repo.name} - Issues`,
+      sourceUrl: repo.url + "/issues",
+      expandable: false,
+      permission: "read",
+      dustDocumentId: null,
+      lastUpdatedAt: null,
+    });
+  });
+  allDiscussionsFromRepoIds.forEach((repoId) => {
+    const repo = uniqueRepos[repoId];
+    if (!repo) {
+      return;
+    }
+    nodes.push({
+      provider: c.type,
+      internalId: `${repoId}-discussions`,
+      parentInternalId: repoId.toString(),
+      type: "channel",
+      title: "Discussions",
+      titleWithParentsContext: `${repo.name} - Discussions`,
+      sourceUrl: repo.url + "/discussions",
+      expandable: false,
+      permission: "read",
+      dustDocumentId: null,
+      lastUpdatedAt: null,
+    });
+  });
+
+  // Constructing Nodes for Code
+  fullCodeInRepos.forEach((codeRepo) => {
+    const repo = uniqueRepos[parseInt(codeRepo.repoId)];
+    nodes.push({
+      provider: c.type,
+      internalId: `github-code-${codeRepo.repoId}`,
+      parentInternalId: codeRepo.repoId,
+      type: "folder",
+      title: "Code",
+      titleWithParentsContext: repo ? `[${repo.name}] - Code` : "Code",
+      sourceUrl: codeRepo.sourceUrl,
+      expandable: true,
+      permission: "read",
+      dustDocumentId: null,
+      lastUpdatedAt: codeRepo.codeUpdatedAt.getTime(),
+    });
+  });
+
+  // Constructing Nodes for Code Directories
+  codeDirectories.forEach((directory) => {
+    const repo = uniqueRepos[parseInt(directory.repoId)];
+    nodes.push({
+      provider: c.type,
+      internalId: directory.internalId,
+      parentInternalId: directory.parentInternalId,
+      type: "folder",
+      title: directory.dirName,
+      titleWithParentsContext: repo
+        ? `[${repo.name}] Code Directory "${directory.dirName}"`
```

Comment:
Sorry late to the party I would just remove "Code Directory" and "Code File" bellow. The type is covered by the icon and the fact that it comes from code almost obvious.

Maybe `[dust] README.md (code)` `[dust] srv (code)` if you care about keeping the code part.

Reply:
https://github.com/dust-tt/dust/pull/4119